### DISCUSSION TITLE
ST75256 i2c bus mode

### DISCUSSION THREAD

**Discussion by elektronek** [olikraus/u8g2#1778] [General]
Hello Olikraus,
thanks for reading, i'm testing display type see URL
https://www.winstar.com.tw/products/cog-lcd-module-display/256x128-lcd.html

driver identified as ST75256, switched in i2c mode, scanner found it at 0x3C address,
but not working, display is empty,  not showed any pixel,  my constructor is below:

#include <Arduino.h>
#include <U8g2lib.h>
#include <Wire.h>

U8G2_ST75256_WO256X128_F_HW_I2C  u8g2(U8G2_R0, /* reset=*/ U8X8_PIN_NONE);

  void setup()
  }
   u8g2.setI2CAddress(0x3C);
   u8g2.begin();
   //  u8g2.setContrast(128);  // try to change contrast
 }

 void loop() {
   u8g2.clearBuffer();					// clear the internal memory
   u8g2.setFont(u8g2_font_ncenB08_tr);	// choose a suitable font
   u8g2.drawStr(10,10,"Hello World!");	// write something to the internal memory
   u8g2.sendBuffer();					// transfer internal memory to the display
   delay(1000);  
 }

Do you have please any suggestions ?
Other devices on this i2c (eeprom on 0x50) working correctly
U8g2 latest stable version, ESP32 board

Best Regards!
Martin

**Comment by olikraus** [comment]
I think you need to use "u8g2.setI2CAddress(0x3C*2);".
As mentioned in the wiki, the i2c address has to be multiplied by 2: https://github.com/olikraus/u8g2/wiki/u8g2reference#seti2caddress

It might be odd and you will find a lot of discussion here in the issues whether the multiplication makes sense or not.

**Reply by elektronek** [reply]
Thank you very much, now working OK

Martin
